### PR TITLE
infoschema: change the way of inspection schema obtaining rows from original table

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

The inspection schema is a proxy schema of some tables in information schema, which will cache some rows in it. The previous implementation of inspection schema assumes that all rows can be retrieved via `it.getRows()` from information schema, but this assumption is broken because some memory tables use retriever to obtain rows and they are implemented in the `executor` package (e.g: `cluster_config`).

### What is changed and how it works?

Unify the way of obtaining rows and use internal SQL.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
```
mysql> select * from inspection_result;
+---------+-----------------------+--------------+------------+----------+-----------------------------------------------------------------------------------------------------+
| RULE    | ITEM                  | VALUE        | REFERENCE  | SEVERITY | SUGGESTION                                                                                          |
+---------+-----------------------+--------------+------------+----------+-----------------------------------------------------------------------------------------------------+
| config  | lease                 | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tidb' and `key`='lease'                 |
| config  | log-file              | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='log-file'              |
| config  | log.file.filename     | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tidb' and `key`='log.file.filename'     |
| config  | port                  | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tidb' and `key`='port'                  |
| config  | raftstore.raftdb-path | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='raftstore.raftdb-path' |
| config  | server.addr           | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='server.addr'           |
| config  | server.advertise-addr | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='server.advertise-addr' |
| config  | server.status-addr    | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='server.status-addr'    |
| config  | status.status-port    | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tidb' and `key`='status.status-port'    |
| config  | storage.data-dir      | inconsistent | consistent | warning  | select * from information_schema.cluster_config where type='tikv' and `key`='storage.data-dir'      |
| version | tidb                  | inconsistent | consistent | critical | select * from information_schema.cluster_info where type='tidb'                                     |
+---------+-----------------------+--------------+------------+----------+-----------------------------------------------------------------------------------------------------+
11 rows in set (0.05 sec)

mysql> select * from information_schema.cluster_config where type='tidb' and `key`='lease';
+------+---------------+-------+-------+
| TYPE | ADDRESS       | KEY   | VALUE |
+------+---------------+-------+-------+
| tidb | 0.0.0.0:49920 | lease | 47s   |
| tidb | 0.0.0.0:49910 | lease | 46s   |
| tidb | 0.0.0.0:49930 | lease | 45s   |
| tidb | 0.0.0.0:49900 | lease | 45s   |
+------+---------------+-------+-------+
4 rows in set (0.00 sec)

mysql> select * from information_schema.cluster_info where type='tidb';
+------+---------------+----------------+------------------------------------------+------------------------------------------+
| TYPE | ADDRESS       | STATUS_ADDRESS | VERSION                                  | GIT_HASH                                 |
+------+---------------+----------------+------------------------------------------+------------------------------------------+
| tidb | 0.0.0.0:49920 | 0.0.0.0:12080  | 5.7.25-TiDB-v4.0.0-alpha-1276-gc1bc9ffe5 | c1bc9ffe5bf69e3e1acc5a9297c410e03b24bcd3 |
| tidb | 0.0.0.0:49910 | 0.0.0.0:11080  | 5.7.25-TiDB-v4.0.0-alpha-1276-gc1bc9ffe5 | c1bc9ffe5bf69e3e1acc5a9297c410e03b24bcd3 |
| tidb | 0.0.0.0:49930 | 0.0.0.0:13080  | 5.7.25-TiDB-v4.0.0-alpha-1277-gb009ca3fd | b009ca3fd7e3c80c7506ce7d425f0f21f8c6409b |
| tidb | 0.0.0.0:49900 | 0.0.0.0:10080  | 5.7.25-TiDB-v4.0.0-alpha-1276-gc1bc9ffe5 | c1bc9ffe5bf69e3e1acc5a9297c410e03b24bcd3 |
+------+---------------+----------------+------------------------------------------+------------------------------------------+
4 rows in set (0.00 sec)
```

### CODE REVIEW

File: infoschema/inspection_schema.go [hunk 1]
```diff
@@ -60,8 +62,19 @@ func (it *inspectionSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv
 	// Obtain data from cache first.
 	cached, found := sessionVars.InspectionTableCache[it.meta.Name.L]
 	if !found {
-		// We retrieve data from `information_schema` if can found in cache.
-		rows, err := it.getRows(ctx, cols)
+		// We retrieve data from `information_schema` if cannot found in cache.
```
Comment (AilinKid) [inline infoschema/inspection_schema.go]:
```suggestion
		// Retrieve data from `information_schema` if cannot found in cache.
```

File: executor/diagnostics.go [hunk 2]
```diff
@@ -138,10 +138,10 @@ func (configInspection) inspect(_ context.Context, sctx sessionctx.Context, filt
 			continue
 		}
 		results = append(results, inspectionResult{
-			item:      row.GetString(1),                   // key
-			value:     fmt.Sprintf("%d", row.GetInt64(2)), // count
-			reference: "1",
-			severity:  "P2",
+			item:      row.GetString(1), // key
+			value:     "inconsistent",
+			reference: "consistent",
```
Comment (djshow832) [inline executor/diagnostics.go]: Since you changed numbers to strings, change the fields to 'actual' and 'expected' may be better.

Comment (lonng) [reply executor/diagnostics.go]: I think it's better to change the `reference` to `expected` but keep the `value` unchanged. The `value` may be applied to more rules properly.

File: infoschema/inspection_schema.go [hunk 3]
```diff
@@ -60,8 +62,19 @@ func (it *inspectionSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv
...
 	cached, found := sessionVars.InspectionTableCache[it.meta.Name.L]
 	if !found {
-		// We retrieve data from `information_schema` if can found in cache.
-		rows, err := it.getRows(ctx, cols)
+		// Retrieve data from `information_schema` if cannot found in cache.
+		sql := "select * from information_schema." + it.meta.Name.L
+		results, fieldTypes, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+		var rows [][]types.Datum
+		if len(results) > 0 {
+			var fileds []*types.FieldType
```
Comment (djshow832) [inline infoschema/inspection_schema.go]:
```suggestion
			fields := make([]*types.FieldType, 0, len(fieldTypes))
```

File: infoschema/inspection_schema.go [hunk 4]
```diff
@@ -60,8 +62,19 @@ func (it *inspectionSchemaTable) IterRecords(ctx sessionctx.Context, startKey kv
 	// Obtain data from cache first.
 	cached, found := sessionVars.InspectionTableCache[it.meta.Name.L]
 	if !found {
-		// We retrieve data from `information_schema` if can found in cache.
-		rows, err := it.getRows(ctx, cols)
+		// Retrieve data from `information_schema` if cannot found in cache.
+		sql := "select * from information_schema." + it.meta.Name.L
+		results, fieldTypes, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+		var rows [][]types.Datum
```
Comment (djshow832) [inline infoschema/inspection_schema.go]:
```suggestion
		rows := make([][]types.Datum, 0, len(results))
```

**Pull request merged**